src/Extension/Traits/FluentObjectTrait.php

```diff
@@ -163,6 +164,38 @@ protected function updateFluentCMSFields(FieldList $fields)
             $this->LinkedLocales(),
             $config
         );
+
+        /** @var GridFieldFilterHeader $filterHeader */
+        $filterHeader = $config->getComponentByType(GridFieldFilterHeader::class);
+
+        // Replace scaffolded filters as the displayed data is in ArrayList format so scaffolded filters do not work
+        if ($filterHeader) {
+            // Retrieve filters settings as these can be carried over as is
+            $defaultSearchContext = $filterHeader->getSearchContext($gridField);
+            $defaultSearchFields = $defaultSearchContext->getSearchFields();
+            $defaultFilters = $defaultSearchContext->getFilters();
+
+            /** @var ArrayList $list */
+            $list = $gridField->getList();
+
+            // Carry over any search form settings
+            $searchContext = BasicSearchContext::create($list->dataClass());
```

Comment:
`dataClass()` is not part of the `SS_List` interface - sorry, didn't notice this until I spun this up locally to test.
```suggestion
            $searchContext = BasicSearchContext::create($gridField->getModelClass());
```